controller/dataplane/owned_deployment.go

```diff
@@ -306,17 +352,32 @@ func reconcileDataPlaneDeployment(
 
 		k8sresources.SetDefaultsPodTemplateSpec(&desired.Spec.Template)
 
+		// Keep track of this for logging purposes
+		originalReplicaCount := existing.Spec.Replicas
+
 		// ensure that object metadata is up to date
 		updated, existing.ObjectMeta = k8sutils.EnsureObjectMetaIsUpdated(existing.ObjectMeta, desired.ObjectMeta)
 
-		// some custom comparison rules are needed for some PodTemplateSpec sub-attributes, in particular
-		// resources and affinity.
+		isRestartOperation := isRecentDeploymentRestart(&existing.Spec.Template, logger)
+
+		// some custom comparison rules are needed for some PodTemplateSpec sub-attributes
 		opts := []cmp.Option{
 			cmp.Comparer(k8sresources.ResourceRequirementsEqual),
+			utils.IgnoreAnnotationKeysComparer(restartAnnotationKey),
 		}
 
-		// ensure that PodTemplateSpec is up to date
 		if !cmp.Equal(existing.Spec.Template, desired.Spec.Template, opts...) {
+
+			if isRestartOperation {
+				// Preserve the restart annotation
+				if desired.Spec.Template.Annotations == nil {
+					desired.Spec.Template.Annotations = make(map[string]string)
+				}
+				// Get the restart timestamp from the existing template
+				restartTimeStr, _ := podTemplateSpecHasRestartAnnotation(&existing.Spec.Template)
```

Comment:
To prevent calling `podTemplateSpecHasRestartAnnotation` here again (it's already called inside `isRecentDeploymentRestart`) we could add a return value to `isRecentDeploymentRestart` to return the annotation value if present. WDYT?

Reply:
Yes, we can avoid the function call here by increasing the return value. Do you have any suggestions regarding the function name?   If the return value is increased, then the name `isRecentDeploymentRestart` will no longer be appropriate.

Reply:
`isRecentDeploymentRestart` I think would do with an appropriate godoc added to it. WDYT?